eslint/babel-eslint-parser/test/typescript-estree.test.js

```diff
@@ -0,0 +1,136 @@
+const { require } = commonJS(import.meta.url);
+
+const PROPS_TO_REMOVE = [
+  // typescript-eslint generates start/end in the loc object, while Babel generate both start/end and loc
+  { key: "start", type: "Node" },
+  { key: "end", type: "Node" },
+
+  { key: "importKind", type: "Node" },
+  { key: "exportKind", type: "Node" },
+  { key: "variance", type: "Node" },
+  { key: "typeArguments", type: "Node" },
+  { key: "filename", type: null },
+  { key: "identifierName", type: null },
+  // For legacy estree AST
+  { key: "attributes", type: "ImportExpression" },
+];
+
+function deeplyRemoveProperties(obj, props) {
+  for (const [k, v] of Object.entries(obj)) {
+    if (
+      props.some(
+        ({ key, type }) =>
+          key === k &&
+          ((type === "Node" && obj.type) || type === obj.type || type == null),
+      )
+    ) {
+      delete obj[k];
+      continue;
+    }
+
+    if (typeof v === "object") {
+      if (Array.isArray(v)) {
+        for (const el of v) {
+          if (el != null) {
+            deeplyRemoveProperties(el, props);
+          }
+        }
+      }
+
+      if (v != null) {
+        deeplyRemoveProperties(v, props);
+      }
+    }
+  }
+}
+
+(isESLint9 ? describe : describe.skip)(
+  "Babel should output the same AST as TypeScript-Estree",
+  () => {
+    /**
+     * @type {import("@typescript-eslint/typescript-estree")}
+     */
+    let tsEstree;
+
+    /**
+     * @type {import("@typescript-eslint/typescript-estree").TSESTreeOptions}
+     */
+    const tsEstreeOptions = {
+      filePath: "typescript-estree.test.js-test-input.tsx",
+      jsx: true,
+      tokens: true,
+      loc: true,
+      range: true,
+      comment: true,
+      sourceType: "module",
+    };
+
+    function parseAndAssertSame(code, babelEcmaFeatures = null) {
+      if (isESLint9) {
+        // ESLint 9
+        const tsEstreeAST = tsEstree.parse(code, tsEstreeOptions);
+        const babelAST = parseForESLint(code, {
+          eslintVisitorKeys: true,
+          eslintScopeManager: true,
+          ecmaFeatures: babelEcmaFeatures,
+          requireConfigFile: false,
+          babelOptions: {
+            configFile: false,
+            parserOpts: {
+              plugins: ["jsx", "typescript"],
+            },
+          },
+        }).ast;
+
+        deeplyRemoveProperties(babelAST, PROPS_TO_REMOVE);
+        expect(babelAST).toEqual(tsEstreeAST);
+      }
+    }
+
+    beforeAll(() => {
+      // Use the version of TS-Espree that is a dependency of
+      // the version of typescript-eslint we are testing against.
+      const tsEstreePath = require.resolve(
+        "@typescript-eslint/typescript-estree",
+        {
+          paths: [path.dirname(require.resolve("typescript-eslint"))],
+        },
+      );
+
+      tsEstree = require(tsEstreePath);
+    });
+
+    it.each([
+      ["empty", ""],
+
+      ["boolean", "true"],
+      ["boolean", "false"],
+
+      ["numeric", "0"],
+      // ["bigint", "0n"],
```

Comment:
This test is commented out due to https://github.com/typescript-eslint/typescript-eslint/issues/11018